src/main/java/de/mediathekview/mserver/crawler/wdr/parser/WdrVideoJsonDeserializer.java

```diff
@@ -0,0 +1,148 @@
+package de.mediathekview.mserver.crawler.wdr.parser;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import de.mediathekview.mserver.base.utils.JsonUtils;
+import de.mediathekview.mserver.crawler.wdr.WdrMediaDTO;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class WdrVideoJsonDeserializer implements JsonDeserializer<Optional<WdrMediaDTO>> {
+  
+  private static final Logger LOG = LogManager.getLogger(WdrVideoJsonDeserializer.class);
+  
+  private static final String ELEMENT_ALT = "alt";
+  private static final String ELEMENT_CAPTION_HASH = "captionsHash";
+  private static final String ELEMENT_DFLT = "dflt";
+  private static final String ELEMENT_MEDIA_RESOURCE = "mediaResource";
+  
+  private static final String ATTRIBUTE_AUDIO_DESCRIPTION = "adVideoURL";
+  private static final String ATTRIBUTE_CAPTION_URL = "captionURL";
+  private static final String ATTRIBUTE_SIGN_LANUAGE = "slVideoURL";
+  private static final String ATTRIBUTE_VERSION = "mediaVersion";
+  private static final String ATTRIBUTE_VIDEO_URL = "videoURL";
+  private static final String ATTRIBUTE_XML = "xml";
+
+  private static final String MEDIA_VERSION_1_1 = "1.1.0";
+  private static final String MEDIA_VERSION_1_2 = "1.2.0";
+  private static final String MEDIA_VERSION_1_3 = "1.3.0";
+  
+  private final String protocol;
+  
+  public WdrVideoJsonDeserializer(final String aProtocol) {
+    protocol = aProtocol;
+  }
+  
+  @Override
+  public Optional<WdrMediaDTO> deserialize(JsonElement aJsonElement, Type aType, JsonDeserializationContext aContext) throws JsonParseException {
+    JsonObject jsonObject = aJsonElement.getAsJsonObject();
+    
+    Optional<String> version = JsonUtils.getAttributeAsString(jsonObject, ATTRIBUTE_VERSION);
+    if (version.isPresent()) {
+      switch(version.get()) {
+        case MEDIA_VERSION_1_1:
+          return deserializeVersion1_1(jsonObject);
+        case MEDIA_VERSION_1_2:
+          return deserializeVersion1_2(jsonObject);
+        case MEDIA_VERSION_1_3:
+          return deserializeVersion1_3(jsonObject);
+        default:
+          LOG.error("WdrVideoJsonDeserializer: unsupported media version: " + version.get());
+      }
+    }
+    
+    return Optional.empty();
+  }  
+
+  private Optional<WdrMediaDTO> deserializeVersion1_1(final JsonObject aJsonObject) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)